utils/mochaRunner/src/main.ts

```diff
@@ -0,0 +1,191 @@
+import os from 'os';
+import {spawn} from 'node:child_process';
+import {
+  extendProcessEnv,
+  filterByPlatform,
+  prettyPrintJSON,
+  readJSON,
+  filterByParameters,
+  getExpectationUpdates,
+  getSkippedTests,
+} from './utils.js';
+
+async function main() {
+  const testSuiteArgIdx = process.argv.indexOf('--test-suite');
+
+  const platform = zPlatform.parse(os.platform());
+
+  const expectations = readJSON(
+    path.join(process.cwd(), 'test', 'TestExpectations.json')
+  ) as TestExpectation[];
+
+  const parsedSuitesFile = zTestSuiteFile.parse(
+    readJSON(path.join(process.cwd(), 'test', 'TestSuites.json'))
+  );
+
+  let applicableSuites: TestSuite[] = [];
+
+  if (testSuiteArgIdx === -1) {
+    applicableSuites = filterByPlatform(parsedSuitesFile.testSuites, platform);
+  } else {
+    const testSuiteId = process.argv[testSuiteArgIdx + 1];
+    const testSuite = parsedSuitesFile.testSuites.find(suite => {
+      return suite.id === testSuiteId;
+    });
+
+    if (!testSuite) {
+      console.error(`Test suite ${testSuiteId} is not defined`);
+      process.exit(1);
+    }
+
+    if (!testSuite.platforms.includes(platform)) {
+      console.warn(
+        `Test suite ${testSuiteId} is not enabled for your platform. Running it anyway.`
+      );
+    }
+
+    applicableSuites = [testSuite];
+  }
+
+  console.log('Planning to run the following test suites', applicableSuites);
+
+  let fail = false;
+  const recommendations = [];
+  try {
+    for (const suite of applicableSuites) {
+      const parameters = suite.parameters;
+
+      const applicableExpectations = filterByParameters(
+        filterByPlatform(expectations, platform),
+        parameters
+      );
+
+      const skippedTests = getSkippedTests(applicableExpectations);
+
+      const env = extendProcessEnv([
+        ...parameters.map(param => {
+          return parsedSuitesFile.parameterDefinitons[param];
+        }),
+        {
+          PUPPETEER_SKIPPED_TEST_CONFIG: JSON.stringify(
+            skippedTests.map(ex => {
+              return ex.testIdPattern;
+            })
+          ),
+        },
+      ]);
+
+      const tmpdir = fs.mkdtempSync(
+        path.join(os.tmpdir(), 'puppeteer-test-runner-')
+      );
+      const tmpfilename = path.join(tmpdir, 'output.json');
+      console.log('Running', JSON.stringify(parameters), tmpfilename);
+      const handle = spawn(
+        'npx mocha',
+        ['--reporter=json', '--reporter-option', 'output=' + tmpfilename],
+        {
+          shell: true,
+          cwd: process.cwd(),
+          stdio: 'inherit',
+          env,
+        }
+      );
+      await new Promise<void>((resolve, reject) => {
+        handle.on('error', err => {
+          reject(err);
+        });
+        handle.on('close', () => {
+          resolve();
+        });
+      });
+      console.log('Finished', JSON.stringify(parameters));
+      try {
+        const results = readJSON(tmpfilename) as MochaResults;
```

Comment:
https://google.github.io/styleguide/tsguide.html#type-assertions-and-object-literals

> Use type annotations (: Foo) instead of type assertions (as Foo) to specify the type of an object literal. This allows detecting refactoring bugs when the fields of an interface change over time.

Reply:
That does not work for assigning unknown type to a type though.